dev/io.openliberty.jakarta.batch.2.1/src/jakarta/batch/runtime/BatchRuntime.java

```diff
@@ -0,0 +1,113 @@
+/*
```

Comment:
Is this a temporary overlay? Where can the original be found? What is the delta that we are adding? Is there an issue open to remove the overlay again in the future?

Reply:
It was an overlay in 2.0 as well.  That is how it works with batch.  @scottkurz can explain about it.  I took the source from the maven repo sources.zip and used the 2.0 version and put the changes that were done in the 2.0 version into the 2.1 version.

Reply:
To answer the rest of the question, most of the class is modified.

Reply:
There may be a way to improve this.  May need to talk with @scottkurz and/or @cgianfra about it.

Reply:
It's been an "overlay" since batch-1.0 back in 2015 or so (EE 7).

The API JAR from Maven Central uses ServiceLoader to find the JobOperator .   In batch-1.0 we decided to use this approach to get a hold of our DS instance rather than I guess exposing it to the ServiceLoader.   I'm sure whoever we asked at the time suggested this approach.  

I'm curious what other APIs/features in Liberty do in similar situations.   We did say clearly though in the Batch 2.1 UFO that it worked this way and that an application could NOT load a different impl using META-INF/services and ServiceLoader.  

It's not something that we expect will change except possibly on a release boundary.

@cgianfra maybe you should review the 2.0 vs. 2.1 batch-api Maven JARs to see if there are any changes we should pull in.

Also the "batch.container.unavailable" message doesn't seem right to me.   What does this have to do with  `<batchPersistence />` not being configured?  Maybe this message has been dragged along since an early prototype?

Reply:
While the copyright holder (IBM) is unlikely to sue in this case ;-), more generally the Apache v2 license states `You must cause any modified files to carry prominent notices stating that You changed the files`.